clock/clock.go

```diff
@@ -0,0 +1,54 @@
+package clock
+
+import (
+	"sync"
+	"time"
+)
+
+// Clock defines an interface for fetching time that may be used instead of the
+// time module.
+type Clock interface {
+	Now() time.Time
+}
+
+// BlockClock delegates calls to the time package.
+type BlockClock struct{}
+
+// NewBlockClock returns a BlockClock that delegates calls to the time package.
+func NewBlockClock() *BlockClock {
+	return &BlockClock{}
+}
+
+// Now returns the current local time.
+func (bc *BlockClock) Now() time.Time {
+	return time.Now()
+}
+
+// MockClock returns a mocked clock implementation that may be manually
+// set for testing things related to time.
+type MockClock struct {
+	nowMu sync.Mutex
+	now   time.Time
+}
+
+// NewMockClock returns a mocked clock implementation that may be manually
+// set for testing things related to time.
+func NewMockClock(n time.Time) *MockClock {
+	return &MockClock{
+		now: n,
+	}
+}
+
+// Now returns the current value of the MockClock.
+func (mc *MockClock) Now() time.Time {
+	mc.nowMu.Lock()
```

Comment:
I am not quite sure what race condition this mutex is guarding against.  Is there one?

Reply:
Yup this was a mistake, removed.